wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubURLParser.java

```diff
@@ -59,6 +59,9 @@ public GithubUrl parse(String url) {
 
     String repoUser = matcher.group("repoUser");
     String repoName = matcher.group("repoName");
+    if (repoName.matches("[\\w-][\\w.-]*?\\.git$")) {
```

Comment:
Hello, I'm not sure we need another regexp there as repoName is already extracted from a regexp

Reply:
I mean, if the expected format is not correct, it needs to be fixed in the first regexp, not there.

Reply:
We're just checking if name is valid and it is contains `.git` part

Reply:
@benoitf we're spent so much time on that but was unable to intergrate all possible variants  into main regexp (since we cant use the string boundaries etc), and it becomes overcomplicated/not readable at all, so decided to split into simple separate regex.